src/main.ts

```diff
@@ -24,6 +27,40 @@ function explodeStrings(input: string): Array<string> {
   return list;
 }
 
+async function createApplicationCredentials(
+  workloadIdentityProvider: string,
+  serviceAccount: string,
+  githubOIDCToken: string,
+) {
+  const rootDir = process.env.GITHUB_WORKSPACE;
+  if (!rootDir) {
+    throw new Error('No path for credentials. Set process.env.GITHUB_WORKSPACE');
+  }
+  // Write github OIDC token
+  const oidcTokenPath = path.join(rootDir, uuidv4());
```

Comment:
We should not use UUIDs for filepaths since it brings in an unnecessary external dependency and still has a chance of conflicting. We should use the primitives in the `fs` package for making a temporary file, since those will automatically retry on conflicts.